staging/src/k8s.io/client-go/tools/clientcmd/validation.go

```diff
@@ -30,11 +30,20 @@ import (
 
 var (
 	ErrNoContext   = errors.New("no context chosen")
-	ErrEmptyConfig = errors.New("no configuration has been provided, try setting KUBERNETES_MASTER environment variable")
+	ErrEmptyConfig = NewEmptyConfigError(errors.New("no configuration has been provided, try setting KUBERNETES_MASTER environment variable"))
 	// message is for consistency with old behavior
 	ErrEmptyCluster = errors.New("cluster has no server defined")
 )
 
+// NewEmptyConfigError returns an error wrapping the given error which IsEmptyConfig() will recognize as an empty config error
+func NewEmptyConfigError(err error) error {
+	return emptyConfigError{err}
+}
+
+type emptyConfigError struct {
+	error
```

Comment:
This seems a bit over complicated, why not just:
```go
type emptyConfigError struct {
    message string
}
```
If we need to embed error in error we'll just use go's `%w` when printing.

Reply:
Just make it similar to `errContextNotFound` below.